src/rust/bitbox02-rust/src/keystore.rs

```diff
@@ -46,6 +46,29 @@ pub fn is_locked() -> bool {
     keystore::_is_locked()
 }
 
+fn verify_seed(encryption_key: &[u8], expected_seed: &[u8]) -> bool {
+    if encryption_key.len() != 32 {
+        return false;
+    }
+
+    let cipher = match bitbox02::memory::get_encrypted_seed_and_hmac() {
+        Ok(cipher) => cipher,
+        Err(_) => return false,
+    };
+    let decrypted = match bitbox_aes::decrypt_with_hmac(encryption_key, &cipher) {
+        Ok(decrypted) => decrypted,
+        Err(_) => return false,
+    };
+    if decrypted.len() != expected_seed.len() {
+        return false;
+    }
+    let mut diff = 0u8;
```

Comment:
I'm surprised you don't just do `expected_seed == decrypted`? Does it need to be a constant time implementation?

Reply:
Yeah I was on the edge on this, the agent just did this proactively and I was wondering if I should change it back. I think I'll change it back, can't see how it helps here.

Reply:
Done